native/src/main/java/io/ballerina/stdlib/data/csvdata/csv/CsvParser.java

```diff
@@ -0,0 +1,1075 @@
+                return StringEscapeUtils.unescapeJava("\\u" + sm.hexBuilder.toString()).charAt(0);
+            }
+
+        }
+
+        private static class HeaderEscapedCharacterProcessingState extends EscapedCharacterProcessingState {
+
+            @Override
+            protected State getSourceState() {
+                return HEADER_ESCAPE_CHAR_STATE;
+            }
+        }
+
+        private static class StringValueEscapedCharacterProcessingState extends EscapedCharacterProcessingState {
+
+            @Override
+            protected State getSourceState() {
+                return STRING_ESCAPE_VALUE_STATE;
+            }
+        }
+
+        private abstract static class EscapedCharacterProcessingState implements State {
+
+            protected abstract State getSourceState();
+
+            @Override
+            public State transition(StateMachine sm, char[] buff, int i, int count) throws CsvParserException {
+                State state = null;
+                char ch;
+                if (i < count) {
+                    ch = buff[i];
+                    sm.processLocation(ch);
+                    if (ch == Constants.LineTerminator.CR) {
+                        CsvUtils.setCarriageTokenPresent(true);
+                    } else if (!(CsvUtils.isCarriageTokenPresent && ch == Constants.LineTerminator.LF)) {
+                        CsvUtils.setCarriageTokenPresent(false);
+                    }
+                    if (ch == EOF) {
+                        handleEndOfTheRow(sm);
+                        return ROW_END_STATE;
+                    }
+                    switch (ch) {
+                        case '"':
+                            sm.append(QUOTES);
+                            state = sm.prevState;
+                            break;
+                        case '\\':
+                            sm.append(REV_SOL);
+                            state = sm.prevState;
+                            break;
+                        case '/':
+                            sm.append(SOL);
+                            state = sm.prevState;
+                            break;
+                        case 'b':
+                            sm.append(BACKSPACE);
+                            state = sm.prevState;
+                            break;
+                        case 'f':
+                            sm.append(FORMFEED);
+                            state = sm.prevState;
+                            break;
+                        case 'n':
+                            sm.append(NEWLINE);
+                            state = sm.prevState;
+                            break;
+                        case 'r':
+                            sm.append(CR);
+                            state = sm.prevState;
+                            break;
+                        case 't':
+                            sm.append(HZ_TAB);
+                            state = sm.prevState;
+                            break;
```

Comment:
Given all of these are the same except for what you append to `sm` would it be possible to combine these cases?. Maybe use a constant map that maps ch to whatever you need to append to `sm`